### ISSUE TITLE
Rewrite BlocklyPropClient (PropellerCommunicator) to allow https

### ISSUE THREAD

**Issue by michel-cf** [parallaxinc/BlocklyProp#810]
Starting January 2017 Google Chrome will start indicating sites with password fields using plain http as unsecure (source: https://security.googleblog.com/2016/09/moving-towards-more-secure-web.html )

To avoid losing users, the application should be changed by then to allow the use https.

**Comment by zfi** [comment]
Perhaps we can remove support for HTTP and use HTTPS exclusively. Does this create any issues with communication to the BlocklyPropClient app?

**Comment by michel-cf** [comment]
That's what we want to do, but then the whole mechanism of the BlocklyPropClient needs to be rewritten.
No local communication between the browser and the BlocklyPropClient is possible if you are working with https.

**Comment by zfi** [comment]
Alright, given the restriction on HTTPS to the BlocklyProp client, what changes are required in the client to overcome the problem? Does the client still use a embedded server? Trying to understand if you are talking about a complete rewrite or a change in the design that requires rewriting some part of the client.

It appears that the beta BlocklyProp site is already configured to use HTTPS. If there is nothing more to change in this project, we can move the discussion to the BlockyPropClient repository.

**Comment by michel-cf** [comment]
We indeed already ran using HTTPS for a short while, then we noticed that BlocklyPropClient didn't work.

I've communicated about my proposed solution before, but it is quite a big change.
The BlocklyPropClient would no longer have an in built server, but instead connect to a server using a websocket. BlocklyPropClient's get authorized to accounts by entering their unique id in a management page in the browser when logged in to the user account.
In the editor, a list of active authorized BlocklyPropClient's is requested from the websocket server (BlocklyPropClient-server). When loading a binary, the binary is sent to the BlocklyPropClient of choice over the active websocket connection.
When opening a serial socket, the browser opens a websocket connection to the BlocklyPropClient-server, indicates the client they want to connect with, and data is passed from one websocket connection to the other.
